### PR TITLE
Add configurable validation for username, name, and email

### PR DESCRIPTION
**Pull request by velicanu**
Hey @mkhorasani  ,

This PR is an alternative implementation to https://github.com/mkhorasani/Streamlit-Authenticator/pull/52 in order to address https://github.com/mkhorasani/Streamlit-Authenticator/issues/53 .

The reasons I propose this implementation (over https://github.com/mkhorasani/Streamlit-Authenticator/pull/52) are the following:
* allows users to specify their own validators if their requirements differ from the defaults here
* loosen default name validation, since names around the world can have all sorts of characters
* loosen email validation, emails are also hard to validate properly, the best strategy is to actually try to email the provided address and wait for a response
* simpler code change to the existing logic

I tested this out in one of my projects using this library and it seems to work.

### CODE REVIEW

File: streamlit_authenticator/validator.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
+import re
+
+
+class Validator:
```
Comment (mkhorasani) [inline streamlit_authenticator/validator.py:3]: Please follow the same commenting format as other classes and functions as shown below:

    def _check_credentials(self, inplace: bool=True) -> bool:
        """
        Checks the validity of the entered credentials.
        Parameters
        ----------
        inplace: bool
            Inplace setting, True: authentication status will be stored in session state, 
            False: authentication status will be returned as bool.
        Returns
        -------
        bool
            Validity of entered credentials.
        """

**Review by mkhorasani — CHANGES_REQUESTED** [review]
Thank you for the changes. Please review the changes/comments and submit them again! Cheers.

**Comment by velicanu** [comment]
> Thank you for the changes. Please review the changes/comments and submit them again! Cheers.

Done: https://github.com/mkhorasani/Streamlit-Authenticator/pull/63/commits/25fc3c552331d617d87c2c80a113e695eb2f5ef7

By the way did you mean to delete the whole setup file or just revert my change in this commit? https://github.com/mkhorasani/Streamlit-Authenticator/pull/63/commits/3ef4a2cf533a8270c0fc365ac4e1562dee8ca791

**Comment by mkhorasani** [comment]
> > Thank you for the changes. Please review the changes/comments and submit them again! Cheers.
> 
> Done: [25fc3c5](https://github.com/mkhorasani/Streamlit-Authenticator/commit/25fc3c552331d617d87c2c80a113e695eb2f5ef7)
> 
> By the way did you mean to delete the whole setup file or just revert my change in this commit? [3ef4a2c](https://github.com/mkhorasani/Streamlit-Authenticator/commit/3ef4a2cf533a8270c0fc365ac4e1562dee8ca791)

That was fast! No, I only meant to revert the changes.

**Review by mkhorasani — APPROVED** [review]

**Comment by mkhorasani** [comment]
Thank you. Pull request merged!

**Comment by velicanu** [comment]
Sure @mkhorasani !

```
        validator: Validator
            A Validator object that checks the validity of the username, name, and email fields
```

**Comment by mkhorasani** [comment]
> Sure @mkhorasani !
> 
> ```
>         validator: Validator
>             A Validator object that checks the validity of the username, name, and email fields
> ```

Cheers!

**Pull request merged**